vendor/compat/string.ml

```diff
@@ -0,0 +1,10 @@
+open Base
+
+let starts_with_whitespace s =
+  (not (String.is_empty s)) && Char.is_whitespace s.[0]
+
+let ends_with_whitespace s =
+  let len = String.length s in
+  len > 1 && Char.is_whitespace s.[len - 1]
```

Comment:
The asymmetry between the two functions is a bit surprising. In particular, this means that `ends_with_whitespace " " = false`.

Would it work to define `ends_with_whitespace` as "not empty and last char is whitespace" (like `starts_with_whitespace`) and move the `len > 1` check to call sites?

Reply:
Yes we can, I wanted to remove the `len > 1` test from the callsites but I can put them back.

Reply:
done